py/selenium/webdriver/chromium/virtual_authenticator.py

```diff
@@ -0,0 +1,248 @@
+# Licensed to the Software Freedom Conservancy (SFC) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The SFC licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from base64 import urlsafe_b64encode, urlsafe_b64decode
+from enum import Enum
+
+
+class Protocol(Enum):
+    """
+    Protocol to communicate with the authenticator.
+    """
+    CTAP2 = "ctap2"
+    U2F = "ctap1/u2f"
+
+
+class Transport(Enum):
+    """
+    Transport method to communicate with the authenticator.
+    """
+    BLE = "ble"
+    USB = "usb"
+    NFC = "nfc"
+    INTERNAL = "internal"
+
+
+class VirtualAuthenticatorOptions:
+
+    Protocol = Protocol
+    Transport = Transport
+
+    def __init__(self) -> None:
+        """Constructor. Initialize VirtualAuthenticatorOptions object.
+
+          :default:
+            - protocol: Protocol.CTAP2
+            - transport: Transport.USB
+            - hasResidentKey: False
+            - hasUserVerification: False
+            - isUserConsenting: True
+            - isUserVerified: False
+        """
+
+        self._protocol: str = Protocol.CTAP2
+        self._transport: str = Transport.USB
+        self._has_resident_key: bool = False
+        self._has_user_verification: bool = False
+        self._is_user_consenting: bool = True
+        self._is_user_verified: bool = False
+
+    @property
+    def protocol(self) -> Protocol:
+        return self._protocol
+
+    @protocol.getter
+    def protocol(self) -> str:
+        return self._protocol.value
+
+    @protocol.setter
+    def protocol(self, protocol: Protocol) -> None:
+        self._protocol = protocol
+
+    @property
+    def transport(self) -> Transport:
+        return self._transport
+
+    @transport.getter
+    def transport(self) -> str:
+        return self._transport.value
+
+    @transport.setter
+    def transport(self, transport: Transport):
+        self._transport = transport
+
+    @property
+    def has_resident_key(self) -> None:
+        return self._has_resident_key
+
+    @has_resident_key.setter
+    def has_resident_key(self, value: bool) -> None:
+        self._has_resident_key = value
+
+    @property
+    def has_user_verification(self):
```

Comment:
minor: types on these would be good; we need to get `mypy` more active in general so feel free to ignore